### PR TITLE
Feature 323/role non blocking

### PR DESCRIPTION
**Pull request by Maroju100**
Updated role controller to non blocking

### CODE REVIEW

File: server/src/main/java/com/objectcomputing/checkins/services/role/RoleController.java [hunk 1]
```diff
@@ -79,8 +110,19 @@
...
+    public Single<HttpResponse<Role>> readRole(UUID id) {
+        return Single.fromCallable(() -> {
+            Role result = roleServices.read(id);
+            if (result == null) {
+                throw new RoleNotFoundException("No role item for UUID");
+            }
+            return result;
+        })
+        .observeOn(Schedulers.from(eventLoopGroup))
+        .map(createdRole -> {
```
Comment (Bagurp) [inline server/src/main/java/com/objectcomputing/checkins/services/role/RoleController.java:122]: Should we rename `createdRole` to improve readability?

File: server/src/main/java/com/objectcomputing/checkins/services/role/RoleController.java [hunk 2]
```diff
@@ -91,9 +133,12 @@ public Role readRole(UUID id) {
...
      */
     @Get("/{?role,memberid}")
-    public Set<Role> findRole(@Nullable RoleType role,
-                              @Nullable UUID memberid) {
-        return roleServices.findByFields(role, memberid);
+    public Single<HttpResponse<Set<Role>>> findRole(@Nullable RoleType role,
+    @Nullable UUID memberid) {
+        return Single.fromCallable(() -> roleServices.findByFields(role, memberid))
+                .observeOn(Schedulers.from(eventLoopGroup))
+                .map(createdRole -> (HttpResponse<Set<Role>>) HttpResponse.ok(createdRole))
```
Comment (Bagurp) [inline server/src/main/java/com/objectcomputing/checkins/services/role/RoleController.java:140]: same here

File: server/src/main/java/com/objectcomputing/checkins/services/role/RoleController.java [hunk 3]
```diff
@@ -60,16 +88,19 @@
...
-        Role updatedRole = roleServices.update(role);
-        return HttpResponse
-                .ok()
-                .headers(headers -> headers.location(
-                        URI.create(String.format("%s/%s", request.getPath(), updatedRole.getId()))))
-                .body(updatedRole);
-
+    public Single<HttpResponse<Role>> update(@Body @Valid Role role, HttpRequest<Role> request) {
+        if (role == null) {
+            return Single.just(HttpResponse.ok());
```
Comment (Bagurp) [inline server/src/main/java/com/objectcomputing/checkins/services/role/RoleController.java:95]: Can `Role` be null if we use `@valid` annotation?

File: server/build.gradle [hunk 4]
```diff
@@ -41,6 +41,8 @@ dependencies {
     annotationProcessor("io.micronaut.configuration:micronaut-openapi:1.5.1")
     annotationProcessor("io.micronaut.security:micronaut-security-annotations")
 
+    compile 'com.google.auth:google-auth-library-oauth2-http:0.21.1'
```
Comment (meyerinj) [inline server/build.gradle:44]: Is this supposed to be included here?

**Review by meyerinj — APPROVED** [review]

File: server/build.gradle (hunk 4 above)
Comment (Maroju100) [reply server/build.gradle]: removed from the file. Thanks.

File: server/src/main/java/com/objectcomputing/checkins/services/role/RoleController.java (hunk 3 above)
Comment (Maroju100) [reply server/src/main/java/com/objectcomputing/checkins/services/role/RoleController.java]: Added @NotNull annotation with @Valid notation and removed unnecessary if null check.

File: server/src/main/java/com/objectcomputing/checkins/services/role/RoleController.java (hunk 1 above)
Comment (Maroju100) [reply server/src/main/java/com/objectcomputing/checkins/services/role/RoleController.java]: updated to userRole for better readability

File: server/src/main/java/com/objectcomputing/checkins/services/role/RoleController.java (hunk 2 above)
Comment (Maroju100) [reply server/src/main/java/com/objectcomputing/checkins/services/role/RoleController.java]: updated to userRole for better readability

**Review by Bagurp — APPROVED** [review]

**Pull request merged**